spotify_tensorflow/dataset.py

```diff
@@ -122,6 +124,20 @@ def get_featran_example_dataset(dir_path,
         num_features, parse_fn = Datasets.get_parse_proto_function(feature_desc_path,
                                                                    feature_mapping_fn)
         dataset = tf.data.Dataset.from_tensor_slices(filenames)
+
+        tf_config = os.environ.get('TF_CONFIG')
+
+        # If TF_CONFIG is not available don't bother sharding
+        if tf_config is not None:
+            tf_config_json = json.loads(tf_config)
+            tf.logging.info('we have a TF_CONFIG: %s' % tf_config)
+            num_workers = len(tf_config_json.get('cluster',{}).get('worker',[]))
+            tf.logging.info('num_workers=%s' % num_workers)
+            worker_index = tf_config_json.get('task', {}).get('index', None)
+            tf.logging.info('sharding on worker_index=%s' % worker_index)
+            if worker_index is not None:
+                dataset = dataset.shard(num_workers, worker_index)
```

Comment:
And you could merge the other two `info` messages here into: `Sharding enabled via worker index %s, out of %s workers`.